service/plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/route/handler/base/DataAggregationRouteBuilder.java

```diff
@@ -100,7 +106,18 @@ public void configure() {
                 .log("destinations ${body}")
                 // The Destinations receiving the Data Collection events get called here.
                 .recipientList().header("DESTINATIONS").delimiter(";")
-                .shareUnitOfWork();
+                .shareUnitOfWork()
+                .to("seda:endOf" + routeId);
+
+        from("seda:endOf" + routeId + "?blockWhenFull=true&size=100")
+                .choice().when(header("DATA_PROCESSING_COMPLETE").isEqualTo(true))
+                .to("seda:processEndOf" + routeId)
+                .endChoice()
+                .end();
+
+        from("seda:processEndOf" + routeId + "?blockWhenFull=true&size=100")
```

Comment:
You don't need to use the "routeId" in this seda endpoint. Sending Scan Statuses does not need to be a dynamic endpoint generated at run time. Refer to the "DataCollectionFileWriteRouteBuilder". It does not need a dynamic route name for every scan type.